pkg/storage/stores/indexshipper/compactor/deletion/delete_requests_manager.go

```diff
@@ -199,52 +227,47 @@ func (d *DeleteRequestsManager) MarkPhaseFailed() {
 	d.deleteRequestsToProcessMtx.Lock()
 	defer d.deleteRequestsToProcessMtx.Unlock()
 
-	d.deleteRequestsToProcess = d.deleteRequestsToProcess[:0]
+	d.deleteRequestsToProcess = map[string]*userDeleteRequests{}
 }
 
 func (d *DeleteRequestsManager) MarkPhaseFinished() {
 	d.deleteRequestsToProcessMtx.Lock()
 	defer d.deleteRequestsToProcessMtx.Unlock()
 
-	for _, deleteRequest := range d.deleteRequestsToProcess {
-		if err := d.deleteRequestsStore.UpdateStatus(context.Background(), deleteRequest.UserID, deleteRequest.RequestID, StatusProcessed); err != nil {
-			level.Error(util_log.Logger).Log(
-				"msg", "failed to mark delete request for user as processed",
-				"delete_request_id", deleteRequest.RequestID,
-				"user", deleteRequest.UserID,
-				"err", err,
-				"deleted_lines", deleteRequest.DeletedLines,
-			)
-		} else {
-			level.Info(util_log.Logger).Log(
-				"msg", "delete request for user marked as processed",
-				"delete_request_id", deleteRequest.RequestID,
-				"user", deleteRequest.UserID,
-				"deleted_lines", deleteRequest.DeletedLines,
-			)
+	for _, userDeleteRequests := range d.deleteRequestsToProcess {
+		for _, deleteRequest := range userDeleteRequests.requests {
+			if err := d.deleteRequestsStore.UpdateStatus(context.Background(), deleteRequest.UserID, deleteRequest.RequestID, StatusProcessed); err != nil {
+				level.Error(util_log.Logger).Log(
+					"msg", "failed to mark delete request for user as processed",
+					"delete_request_id", deleteRequest.RequestID,
+					"user", deleteRequest.UserID,
+					"err", err,
+					"deleted_lines", deleteRequest.DeletedLines,
+				)
+			} else {
+				level.Info(util_log.Logger).Log(
+					"msg", "delete request for user marked as processed",
+					"delete_request_id", deleteRequest.RequestID,
+					"user", deleteRequest.UserID,
+					"deleted_lines", deleteRequest.DeletedLines,
+				)
+			}
+			d.metrics.deleteRequestsProcessedTotal.WithLabelValues(deleteRequest.UserID).Inc()
 		}
-		d.metrics.deleteRequestsProcessedTotal.WithLabelValues(deleteRequest.UserID).Inc()
 	}
 }
 
-func (d *DeleteRequestsManager) IntervalMayHaveExpiredChunks(interval model.Interval, userID string) bool {
+func (d *DeleteRequestsManager) IntervalMayHaveExpiredChunks(_ model.Interval, userID string) bool {
```

Comment:
If we don't need this arg, why are we passing it?

Reply:
In general I think this method needs to be renamed since it seems it no longer deals with an interval

Reply:
The function satisfies an interface which is shared with the retention code.

Reply:
Gotcha!